### PR TITLE
add support for measure parameter defaults

### PR DESCRIPTION
**Pull request by deekim**
these changes allow for parameter defaults that are specified at the measure level
to be utilized during `cql_engine` evaluation.

---

- [WFFHCOHORT-283](https://jsw.ibm.com/browse/WFFHCOHORT-283)

### CODE REVIEW

**Comment by deekim** [comment]
@CoreySandersOH I didn't forget about the discussion we had last week around refactor things to us a "decorator" pattern.
I kept them as `todo`s in a0014e9f83a0e20413895ebdbbf5d93e716ac190.
But since I was out sick the past couple of days, I figured that could be something I circle back to in a different _PR_.

File: cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java [hunk 1]
```diff
@@ -172,7 +193,25 @@ public MeasureReport evaluatePatientMeasure(Measure measure, String patientId, S
...
 		}
 
+		measure.getExtension().stream()
+				.filter(this::isParameterExtension)
+				.forEach(measureDefault ->
+						         seed.getContext().setParameter(
+								         null,
+								         measureDefault.getId(),
+								         toCqlObject(measureDefault.getValue())));
+
```
Comment (csandersdev) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java:203]: I think this is in the wrong place. This should occur _before_ the part were we set the parameters based on the parameters map provided in method invocation. The map is the override values. We want those applied after the defaults specified in the Measure resource.

Comment (deekim) [reply cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java:203]: ah, yep. good call.

**Review by msargentibm — APPROVED** [review]
Seems sensible to me

**Review by shelleyMurphy — APPROVED** [review]

**Pull request merged**